fuel-poa-coordinator/tests/test_trigger.rs

```diff
@@ -130,15 +117,17 @@ impl BlockProducer for MockBlockProducer {
     }
 }
 
+// TODO: The same code is in the `adapters::transaction_selector::select_transactions`. We need
+//  to move transaction selection logic into `TxPool` to avoid duplication of the code in tests.
```

Comment:
It may make more sense to abstract these fns as a separate module, since these tests use a mocked tx pool and shouldn't take a direct dep on the txpool crate.